src/sagemaker_training/smdataparallel.py

```diff
@@ -241,6 +248,18 @@ def _get_instance_type(self):
         logger.info("instance type: %s" % instance_type)
         return instance_type
 
+    def use_smddp_collectives(self):
+        """Check if communication_backend is set to auto.
+        If yes, enable SMDDP collectives support."""
+        sm_training_env = json.loads(self._env_vars.get("SM_TRAINING_ENV"))
+        comm_backend = sm_training_env.get("additional_framework_parameters").get(
+            "sagemaker_communication_backend"
+        )
```

Comment:
can you add a logger statement to dump the `sagemaker_communication_backend` ? This could help us catch if the entry is not set correctly through pysdk.